lib/mapping/CCampaignHandler.cpp

```diff
@@ -151,6 +180,257 @@ std::string CCampaignHandler::readLocalizedString(CBinaryReader & reader, std::s
 	return VLC->generaltexth->translate(stringID.get());
 }
 
+CCampaignHeader CCampaignHandler::readHeaderFromJson(JsonNode & reader, std::string filename, std::string modName, std::string encoding )
+{
+	CCampaignHeader ret;
+
+	ret.version = reader["version"].Integer();
+	if(ret.version < CampaignVersion::VCMI_MIN || ret.version > CampaignVersion::VCMI_MAX)
+	{
+		logGlobal->info("VCMP Loading: Unsupported campaign %s version %d", filename, ret.version);
+		return ret;
+	}
+	
+	ret.version = CampaignVersion::VCMI;
+	ret.mapVersion = reader["campaignId"].Integer();

```

Comment:
IMO we should eventually use mapVersion only for h3c's

And for vcmi campaigns embed all corresponding data (e.g. all names of images) into vcmp file.

This is needed to support Heroes Chronicles, HotA, and likely to be requested feature by map makers.

Reply:
this feature is out of this PR, because needs changes outside parser, in vcmiclient.
For now the workaround is to add campaign_regions.json to the mod with overrides